website/src/components/comment-section.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { ReactNode, SetStateAction } from "react"
+import { useState } from "react"
+import * as Dailp from "src/graphql/dailp"
+import { TranslatedParagraph } from "src/segment"
+import * as css from "./comment-section.css"
+
+export const CommentSection = (p: {
+  parent: Dailp.FormFieldsFragment | TranslatedParagraph
+}) => {
+  return (
+    <div>
+      {p.parent.__typename === "DocumentParagraph" ? (
+        <ParagraphCommentSection paragraph={p.parent} />
+      ) : p.parent.__typename === "AnnotatedForm" ? (
+        <WordCommentSection word={p.parent} />
+      ) : (
+        ""
+      )}
+    </div>
+  )
+}
+
+export const WordCommentSection = (p: { word: Dailp.FormFieldsFragment }) => {
+  const [{ data }] = Dailp.useWordCommentsQuery({
+    variables: { wordId: p.word.id },
+  })
+
+  const wordComments = data?.wordById
+
+  return (
+    <div>
+      {wordComments?.comments.map((comment) => (
+        <div>
+          <CommentHeader comment={comment} />
+          <CommentBody comment={comment} />
+        </div>
+      ))}
+    </div>
+  )
+}
+
+export const ParagraphCommentSection = (p: {
+  paragraph: TranslatedParagraph
+}) => {
+  const [{ data }] = Dailp.useParagraphCommentsQuery({
+    variables: { paragraphId: p.paragraph.id },
+  })
+
+  const paragraphComments = data?.paragraphById
+
+  return (
+    <div>
+      {paragraphComments?.comments.map((comment) => (
+        <div>
+          <div>
+            <CommentHeader comment={comment} />
+          </div>
+          <CommentBody comment={comment} />
+        </div>
+      ))}
+    </div>
+  )
+}
+
+export const CommentBody = (p: { comment: Dailp.Comment }) => {
+  const commentTypeNames: Record<Dailp.CommentType, string> = {
+    // ... TS will then make sure you have an entry for everything on the "CommentTag" type that you import from the codegen
+    [Dailp.CommentType.Story]: "Story",
+    [Dailp.CommentType.Suggestion]: "Suggestion",
+    [Dailp.CommentType.Question]: "Question",
+  }
+
+  return (
+    <div className={css.commentWrapper}>
+      {p.comment.textContent}
+      <div className={css.tagPadding}>
+        <div
+          className={
+            p.comment.commentType === Dailp.CommentType.Story
+              ? css.tagColorStory
+              : p.comment.commentType === Dailp.CommentType.Suggestion
+              ? css.tagColorSuggestion
+              : css.tagColorQuestion
```

Comment:
One thing to consider: You can have named style variants that share some common styling but diverge based on a parameter. Using a variant might clean up things here and lead to less redundant css.

read more: https://vanilla-extract.style/documentation/api/style-variants/

Reply:
Hmm, I'm trying this and it's fine to define a variant but I'm having trouble actually passing in the values I need to use it from the component without getting errors. If this is essential I can keep trying to figure out how to do it but I've spent a while on it and am struggling